### PR TITLE
clementine: Add Spotify buildoption and fix musl

### PR DESCRIPTION
**Pull request by Johnnynator**

### CODE REVIEW

File: srcpkgs/clementine/template [hunk 1]
```diff
@@ -18,13 +18,28 @@ wrksrc="Clementine-$version"
...
 checksum=f885931a9ab7c88607d07b50c64fcce46fc05f13dd2c0a04188c94eff938f37c
 
+if [ -n "$CROSS_BUILD" ]; then
+	hostmakedepends+=" qt-host-tools qt-devel protobuf-c"
+fi
+
+build_options="spotify"
+
+case "$XBPS_TARGET_MACHINE" in
+	x86_64 | i686 | armv6l | armv7l)
```
Comment (Vaelatern) [inline srcpkgs/clementine/template]: missing aarch64? Did you perhaps to make a `*-musl` and a `*`?

**Comment by Vaelatern** [comment]
Not sure why your patch replaces `NULL` with a compatible alternative? This seems like a patch for upstream, that does not affect usability in the slightest.

**Comment by Johnnynator** [comment]
It is required for musl, otherwise I get a error when compiling.

File: srcpkgs/clementine/template (hunk 1 above)
Comment (Johnnynator) [reply srcpkgs/clementine/template]: It is due to the fact that the libspotify package is only for these Arches

**Comment by ebfe** [comment]
broken on armv6 https://build.voidlinux.eu/builders/armv6l_builder/builds/5092/steps/shell_3/logs/stdio

**Comment by ebfe** [comment]
also not sure if enabling a build_option that depends on a nonfree package is a good idea.

**Comment by Johnnynator** [comment]
@ebfe Spotify was the default before so I didn't want to change it and also it can be installed without the non free part.

**Pull request merged**